File: tester/harmony/rnoh/src/main/ets/RNOHCorePackage/turboModules/BlobTurboModule.ts
import { TurboModule, TurboModuleContext } from "../../RNOH/TurboModule";

export class BlobTurboModule extends TurboModule {
  public static readonly NAME = 'BlobModule';

  constructor(protected ctx: TurboModuleContext) {
    super(ctx);
  }

  addNetworkingHandler() {
    console.log('BlobTurboModule call addNetworkingHandler');
  }
}

File: tester/harmony/rnoh/src/main/cpp/RNOHCorePackage/TurboModules/BlobTurboModule.cpp
// NOTE: This entire file should be codegen'ed.

#include "BlobTurboModule.h"

using namespace rnoh;
using namespace facebook;

static jsi::Value __hostFunction_BlobTurboModule_addNetworkingHandler(
    jsi::Runtime &rt,
    react::TurboModule &turboModule,
    const jsi::Value *args,
    size_t count) {
    return static_cast<ArkTSTurboModule &>(turboModule).call(rt, "addNetworkingHandler", args, count);
}

BlobTurboModule::BlobTurboModule(const ArkTSTurboModule::Context ctx, const std::string name)
    : ArkTSTurboModule(ctx, name) {
    methodMap_["addNetworkingHandler"] = MethodMetadata{0, __hostFunction_BlobTurboModule_addNetworkingHandler};
}